### PR TITLE
feat(electron-8x): update electron from 7.0.1 to 8.2.2 

### PR DESCRIPTION
**Pull request by karanbirsingh**
#### Description of changes

This PR updates our electron version from 7.0.1 to 8.2.2. It includes the following changes:

**build pipeline:** As outlined in #2538, for now the environment variables used to download the electron mirror explicitly include the version number to allow for multiple version->mirror mappings in our build pipeline definition. I also update a reference from `devDependencies` to `dependencies` to reflect our updated placement of electron so the windows e2e tests in the release pipeline use the electron mirror. I have a [separate branch in progress](https://github.com/microsoft/accessibility-insights-web/tree/explicit-electron-download) (not part of this feature) that should simplify this process in the future.

**electron version update:** Moving from 7.x to 8.x didn't require any product code changes for us. @DaveTryon is looking ahead at required changes for electron 9.

**spectron update:** Each major version of electron has an associated major spectron version. However, spectron 10.0.1 explicitly adds @types/webdriverio 4.8. Beforehand, these typings were omitted from spectron's production dependencies, so we worked around the untyped library. [@types/webdriverio is deprecated now](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/25186) because webdriverio ships improved types itself. This update required some changes to our e2e tests to play nice with the typings - in particular:
- we add missing promise return types
- we await find-element ($, $$, element) calls
- this works around empty intersection reduction checks introduced in TypeScript 3.5 that affect original typings [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2930acb29bb477facaeb8bc3184e29d129fb3fec/types/webdriverio/index.d.ts#L904) `(string & null) -> never`

spectron update alternatives:
- leaving our code as-is & convincing yarn not to install @types/webdriverio, but this required either an exhaustive types list in tsconfig or a postinstall script
- leaving the client usage untouched & instead adding to webdriver/spectron d.ts files. This worked well, but the explicit `SpectronAsyncClient` interface helps ensure e2e test writers don't accidentally try to await a sync-typed method
- making client private & exposing methods on view-controller. We use enough methods on the client that the new interface made more sense.

---
**validation**: 
- you can see a preliminary unified-e2e-reliability test [here](https://dev.azure.com/accessibility-insights/accessibility-insights-web/_build/results?buildId=8378&view=results).  
- you can find an unsigned installer for this branch [here](https://dev.azure.com/accessibility-insights-private/Accessibility%20Insights%20(private)/_build/results?buildId=9885&view=artifacts&type=publishedArtifacts). I didn't see any UI differences between canary & the new build
- unified-e2e-reliability & web-e2e-reliability PR checks passed
- [failure](https://dev.azure.com/accessibility-insights/accessibility-insights-web/_build/results?buildId=8411&view=logs&j=2fc7ccea-218d-5ea3-c7b2-109cce25cbb6&t=838fae39-ecce-589d-2b59-4936a2881a2d) in initial web-PR in unified-mac job
---

#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [ ] Ran `yarn fastpass`
- [ ] Added/updated relevant unit test(s) (and ran `yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: pipeline/scripts/download-electron-mirror.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ const fs = require('fs');
 
 download(
     {
-        version: pkg.devDependencies.electron,
+        version: pkg.dependencies.electron,
```
Comment (karanbirsingh) [inline pipeline/scripts/download-electron-mirror.js:10]: This is unrelated - we recently moved electron to dependencies from devDependencies and missed this script.

File: src/tests/electron/setup/timeouts.ts [hunk 2]
```diff
@@ -2,11 +2,11 @@
 // Licensed under the MIT License.
 
 // How long to allow an entire test to execute, including startup/setup
-export const DEFAULT_ELECTRON_TEST_TIMEOUT_MS = 30000;
+export const DEFAULT_ELECTRON_TEST_TIMEOUT_MS = 60000;
```
Comment (karanbirsingh) [inline src/tests/electron/setup/timeouts.ts:5]: unrelated to electron update - this loosens some of the values set in https://github.com/microsoft/accessibility-insights-web/pull/2506/files; on my laptop the original values are aggressive enough that the app doesn't usually connect in time.

File: src/tests/electron/common/view-controllers/app-controller.ts [hunk 3]
```diff
@@ -56,13 +56,8 @@ export class AppController {
...
         await this.client.waitUntil(
             async () => {
-                const classes = await this.client.$('body').getAttribute('class');
-
-                if (expectedHighContrastMode) {
-                    return classes.includes(highContrastThemeClass);
-                } else {
-                    return !classes.includes(highContrastThemeClass);
-                }
+                const classes = await this.client.getAttribute<string>('body', 'class');
```
Comment (karanbirsingh) [inline src/tests/electron/common/view-controllers/app-controller.ts:59]: unrelated to update

File: src/tests/electron/common/view-controllers/app-controller.ts [hunk 4]
```diff
@@ -56,13 +56,8 @@ export class AppController {
...
+                return expectedHighContrastMode === classes.includes(highContrastThemeClass);
```
Comment (DaveTryon) [inline src/tests/electron/common/view-controllers/app-controller.ts:60]: Nice reduction!

**Comment by DaveTryon** [comment]
LGTM, but we should get another reviewer before merging
